lib/capability/capability.go

```diff
@@ -0,0 +1,94 @@
+package capability
+
+import (
+	"fmt"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+const CapabilityAnnotation = "capability.openshift.io/name"
+
+type ClusterCapabilities struct {
+	KnownCapabilities map[configv1.ClusterVersionCapability]struct{}
+
+	// value indicates whether capability was explicitly included
+	EnabledCapabilities map[configv1.ClusterVersionCapability]bool
```

Comment:
I'd have expected vendoring [`ClusterVersionCapabilitiesStatus`][1].  Is the local type with the `map` properties because you want a place to hold local state so you could explain tainting?  If so, possibly `map[configv1.ClusterVersionCapability]error` where `error` is `nil` for "spec asked for this" and non-nil for other enabled capabilities, explaining why the property was tainted in?

[1]: https://github.com/openshift/api/blob/7e3ffb09accd36fb0536fa0e69bed5d70cccd6e5/config/v1/types_cluster_version.go#L292-L305

Reply:
Originally, yes, it was to be able to distinguish implicitly enabled caps but it's no longer needed since that will come later. And yeah, I see it becoming whatever we need to report.
And use of maps for easy lookup.